packages/core/src/types/Vector.ts

```diff
@@ -93,6 +94,91 @@ export class Vector2 implements Type {
       Vector2.arcLerp(from, to, value, reverse, ratio);
   }
 
+  /**
+   * Interpolates between two vectors on the polar plane by interpolating
+   * the angles and magnitudes of the vectors individually.
+   *
+   * @param from - The starting vector.
+   * @param to - The target vector.
+   * @param value - The t-value of the interpolation.
+   * @param counterclockwise - Whether the vector should get rotated
+   *                           counterclockwise. Defaults to `false`.
+   * @param origin - The center of rotation. Defaults to the origin.
+   *
+   * @remarks
+   * This function is useful when used in conjunction with {@link rotate} to
+   * animate an object's position on a circular arc (see examples).
+   *
+   * @example
+   * ```tsx
+   * // Animating an object in a circle around the origin
+   * circle().position(
+   *   circle().position().rotate(180),
+   *   1,
+   *   easeInOutCubic,
+   *   Vector2.polarLerp
+   * );
+   *
+   * // Rotating an object around the point [-200, 100]
+   * circle().position(
+   *   circle().position().rotate(180, [-200, 100]),
+   *   1,
+   *   easeInOutCubic,
+   *   Vector2.createPolarLerp(false, [-200, 100]),
+   * );
+   *
+   * // Rotating an object counterclockwise around the origin
+   * circle().position(
+   *   circle().position().rotate(180),
+   *   1,
+   *   easeInOutCubic,
+   *   Vector2.createPolarLerp(true),
+   * );
+   * ```
```

Comment:
Multiple examples should be provided using multiple `@example` tags ([I'm not making this up](https://tsdoc.org/pages/tags/example/)):
```suggestion
   * @example
   * Animating an object in a circle around the origin
   * ```tsx
   * circle().position(
   *   circle().position().rotate(180),
   *   1,
   *   easeInOutCubic,
   *   Vector2.polarLerp
   * );
   * ```
   * @example
   * Rotating an object around the point [-200, 100]
   * ```ts
   * circle().position(
   *   circle().position().rotate(180, [-200, 100]),
   *   1,
   *   easeInOutCubic,
   *   Vector2.createPolarLerp(false, [-200, 100]),
   * );
   * ```
   * @example
   * Rotating an object counterclockwise around the origin
   * ```ts
   * circle().position(
   *   circle().position().rotate(180),
   *   1,
   *   easeInOutCubic,
   *   Vector2.createPolarLerp(true),
   * );
   * ```
```